calico_node/tests/st/policy/test_felix_gateway.py

```diff
@@ -140,216 +154,285 @@ def tearDownClass(cls):
         for host in cls.hosts:
             host.cleanup()
             del host
+        cls.calinet.delete()
 
         log_and_run("docker rm -f cali-st-ext-nginx || true")
 
-    def test_ingress_policy_can_block_through_traffic(self):
-        self.add_policy({
-            'apiVersion': 'v1',
-            'kind': 'policy',
-            'metadata': {'name': 'port80-int'},
-            'spec': {
-                'order': 10,
-                'ingress': [
-                    {
-                        'protocol': 'tcp',
-                        'destination': {'ports': [80]},
-                        'action': 'deny'
-                    },
-                ],
-                'egress': [
-                    {'action': 'deny'},
-                ],
-                'selector': 'role == "gateway-int"'
-            }
-        })
-        self.add_gateway_internal_iface()
-        retry_until_success(self.assert_host_can_not_curl_ext, 3)
+    def test_can_connect_by_default(self):
+        retry_until_success(self.assert_host_can_curl_local, 3)
+        retry_until_success(self.assert_gateway_can_curl_ext, 3)
+        retry_until_success(self.assert_host_can_curl_ext, 3)
+        retry_until_success(self.assert_hostwl_can_access_workload, 3)
+        retry_until_success(self.assert_workload_can_curl_ext, 3)
+
+        self.add_host_iface()
 
-    def test_ingress_policy_can_allow_through_traffic(self):
+        # Adding the host endpoints should break connectivity until we add policy back in.
+        # Add allow policy for host, make sure it applies to forward and has order lower than empty forward.
         self.add_policy({
             'apiVersion': 'v1',
             'kind': 'policy',
-            'metadata': {'name': 'port80-int'},
+            'metadata': {'name': 'host-out'},
             'spec': {
-                'order': 10,
-                'ingress': [
-                    {
-                        'protocol': 'tcp',
-                        'destination': {'ports': [80]},
-                        'action': 'allow'
-                    },
-                ],
-                'egress': [
-                    {'action': 'deny'},
-                ],
-                'selector': 'role == "gateway-int"'
+                'order': 100,
+                'selector': 'nodeEth == "host"',
+                'egress': [{'action': 'allow'}],
+                'ingress': [{'action': 'allow'}],
+                'applyOnForward': True,
             }
         })
+        retry_until_success(self.assert_host_can_curl_ext, 3)
```

Comment:
You're only testing host->ext here.  I think there should be 5 checks, like you're doing everywhere else